### ISSUE TITLE
Photo Gallery with a customizable content overlay for Testimonials, Reviews and How To's

### ISSUE THREAD

**Issue by kartzke** [soflyy/oxygen-bugs-and-features#3272]
**Describe the feature you'd like to see included in Oxygen.**
I would love to have an Oxygen element that allowed users to create an image gallery experience with support for rich text, icons, shortcodes, CTA's and other nested elements.  

When users click or tap to see the full screen image, they would also see this rich content in the lightbox.  On mobile, the content probably needs to be an expand/collapse feature, since the amount of content is variable and could cover the entire image on small screens.

This would be a huge upgrade from the current `captions` feature found in the Oxygen Image Gallery.

For my use case, I'd like to attach review content and testimonials to specific photos.  I'd be very happy with a Photo Testimonials element.  However, this could be used for other things like step-by-step instructions (recipes, how-to's, etc)

So in summary, each image would have a customizable content panel that would allow for the display of content like:

- Title text
- Avatar images
- Star ratings
- Rich text
- Forms and buttons
- Shortcodes

**What are the use cases for this feature?**
- eCommerce
- Photo Testimonials
- How To's and Step-by-Step instructions

**Examples of this feature or functionality.**
I haven't seen any Wordpress components like this, although they might exist.  You can see photo testimonials on a lot of review sites, like Yelp, Google and TripAdvisor.
